test/test_psqlgraph.py

```diff
@@ -1318,10 +1321,123 @@ def test_library_functions_use_session_implicitly(self):
             g.node_lookup(node_id=id1).one()
 
 
+def no_allowed_2_please(edge):
+    if not isinstance(edge.src, Foo):
+        return True
+
+    node = edge.src
+    if node.baz == 'allowed_2':
+        return False
+
+    return True
+
+
+def set_sysan_flag(node):
+    node.sysan['sysan_flag'] = True
+    flag_modified(node, '_sysan')
+
+
+class TestPsqlGraphTraversal(BasePsqlGraphTestCase):
+
+    def setUp(self):
+        super(TestPsqlGraphTraversal, self).setUp()
+
+        with g.session_scope() as session:
+            root_node = FooBar(node_id=str(uuid.uuid4()), bar='root')
+
+            foo1 = Foo(node_id=str(uuid.uuid4()), bar='foo1', baz='allowed_2')
+            foo2 = Foo(node_id=str(uuid.uuid4()), bar='foo2', baz='allowed_1')
+            foo3 = Foo(node_id=str(uuid.uuid4()), bar='foo3', baz='allowed_1')
+
+            test1 = Test(node_id=str(uuid.uuid4()), key1='test1')
+            test2 = Test(node_id=str(uuid.uuid4()), key1='test2')
+            test3 = Test(node_id=str(uuid.uuid4()), key1='test3')
+
+            for foo in [foo1, foo2, foo3]:
+                root_node.foos.append(foo)
+
+            for test in [test1, test2]:
+                foo1.tests.append(test)
+
+            foo2.tests.append(test3)
+
+            session.merge(root_node)
+
+        self.sysan_flag_nodes = [root_node, foo2, foo3, test3]
+        self.not_sysan_flag_nodes = [foo1, test1, test2]
+
+    def test_default_traversal(self):
+        with g.session_scope():
+            root = g.nodes(FooBar).first()
+            traversal = [n for n in root.bfs_children()]
+
+            nodes_all_set = {n.node_id for n in g.nodes().all()}
+
+        self.assertEqual({n.node_id for n in traversal}, nodes_all_set)
+
+    def test_default_traversal_with_transform(self):
```

Comment:
Do you even need this test case? Since you get node from the traversal generator, I'm thinking  the "transform" is not part of the traversal anymore. You're kind of testing the set_sysan_flag() here.

Reply:
agree